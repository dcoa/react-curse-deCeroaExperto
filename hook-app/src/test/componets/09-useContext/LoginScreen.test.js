const { mount } = require("enzyme");
const { LoginScreen } = require("../../../components/09-useContext/LoginScreen");
const { UserContext } = require("../../../components/09-useContext/UserContext");

describe('Test on LoginScreen', () => {

    const setUser = jest.fn();

    const wrapper = mount(
        <UserContext.Provider
            value={{
                user: {
                    name: 'Jhon Doe',
                    email: 'example@fff.com'
                },
                setUser
            }}>
            <LoginScreen />
        </UserContext.Provider>
    );

    test('should show correctly', () => {

        expect(wrapper).toMatchSnapshot();
    });

    test('should call setUser with the arguments', () => {

        wrapper.find('button').simulate('click');
        expect(setUser).toBeCalledWith({
            id: expect.any(Number),
            name: 'Pikachu',
            type: 'Electric'
        });
    });

})
