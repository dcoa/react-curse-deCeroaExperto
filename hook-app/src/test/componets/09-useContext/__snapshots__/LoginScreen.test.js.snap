// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on LoginScreen should show correctly 1`] = `
<div>
  <h1>
    Log in
  </h1>
  <hr />
  <button
    className="btn btn-warning"
    onClick={[Function]}
  >
    Log in
  </button>
</div>
`;
